src/libs/OptionsListUtils.ts

```diff
@@ -1425,69 +1390,43 @@ function getSearchOptions(reports, personalDetails, searchValue = '', betas) {
 
 /**
  * Build the IOUConfirmation options for showing the payee personalDetail
- *
- * @param {Object} personalDetail
- * @param {String} amountText
- * @returns {Object}
  */
-function getIOUConfirmationOptionsFromPayeePersonalDetail(personalDetail, amountText) {
-    const formattedLogin = LocalePhoneNumber.formatPhoneNumber(personalDetail.login);
+function getIOUConfirmationOptionsFromPayeePersonalDetail(personalDetail: PersonalDetails, amountText: string): PayeePersonalDetails {
+    const formattedLogin = LocalePhoneNumber.formatPhoneNumber(personalDetail.login ?? '');
     return {
-        text: personalDetail.displayName || formattedLogin,
-        alternateText: formattedLogin || personalDetail.displayName,
+        text: personalDetail.displayName ? personalDetail.displayName : formattedLogin,
+        alternateText: formattedLogin ?? personalDetail.displayName,
         icons: [
             {
                 source: UserUtils.getAvatar(personalDetail.avatar, personalDetail.accountID),
-                name: personalDetail.login,
+                name: personalDetail.login ?? '',
                 type: CONST.ICON_TYPE_AVATAR,
                 id: personalDetail.accountID,
             },
         ],
         descriptiveText: amountText,
-        login: personalDetail.login,
+        login: personalDetail.login ?? '',
         accountID: personalDetail.accountID,
     };
 }
 
 /**
  * Build the IOUConfirmationOptions for showing participants
- *
- * @param {Array} participants
- * @param {String} amountText
- * @returns {Array}
  */
-function getIOUConfirmationOptionsFromParticipants(participants, amountText) {
-    return _.map(participants, (participant) => ({
+function getIOUConfirmationOptionsFromParticipants(participants: Participant[], amountText: string) {
+    return participants.map((participant) => ({
         ...participant,
         descriptiveText: amountText,
     }));
 }
 
 /**
  * Build the options for the New Group view
- *
- * @param {Object} reports
- * @param {Object} personalDetails
- * @param {Array<String>} [betas]
- * @param {String} [searchValue]
- * @param {Array} [selectedOptions]
- * @param {Array} [excludeLogins]
- * @param {Boolean} [includeOwnedWorkspaceChats]
- * @param {boolean} [includeP2P]
- * @param {boolean} [includeCategories]
- * @param {Object} [categories]
- * @param {Array<String>} [recentlyUsedCategories]
- * @param {boolean} [includeTags]
- * @param {Object} [tags]
- * @param {Array<String>} [recentlyUsedTags]
- * @param {boolean} [canInviteUser]
- * @param {boolean} [includeSelectedOptions]
- * @returns {Object}
  */
 function getFilteredOptions(
-    reports,
-    personalDetails,
-    betas = [],
+    reports: Record<string, Report>,
+    personalDetails: OnyxCollection<PersonalDetails>,
```

Comment:
```suggestion
    reports: OnyxCollection<Report>,
    personalDetails: OnyxEntry<PersonalDetailsList>,
```

`reports` should be an Onyx collection.

PersonalDetails is not a Onyx Collection.

Reply:
Add return type also.